### PR TITLE
book: added section for using finalizers

### PR DESCRIPTION
**Pull request by droot**

### CODE REVIEW

File: docs/book/beyond_basics/using_finalizers.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	// name of your custom finalizer
+	myFinalizerName := "storage.finalizers.example.com"
+
+	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object.
+		if !containsString(instance.ObjectMeta.Finalizers, myFinalizerName) {
+			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, myFinalizerName)
+			if err := r.Update(context.Background(), instance); err != nil {
+				return reconcile.Result{}, err
```
Comment (pwittrock) [inline docs/book/beyond_basics/using_finalizers.md]: Should we return?  Should we re-enqueue?

**Comment by daviddyball** [comment]
@droot One comment, the generated unittests for this fail when finalizer code is implemented. I think it's because `Client.Delete()` needs to get called twice, once to remove a finalizer and once to remove the `CustomResource` itself.

Would it be worth putting a note in the docs to explain this?

File: docs/book/beyond_basics/using_finalizers.md (hunk 1 above)
Comment (droot) [reply docs/book/beyond_basics/using_finalizers.md]: Good catch, it should be re-queue.

Comment (droot) [reply docs/book/beyond_basics/using_finalizers.md]: Done.

**Comment by droot** [comment]
@daviddyball 
>One comment, the generated unittests for this fail when finalizer code is implemented. I think it's because Client.Delete() needs to get called twice, once to remove a finalizer and once to remove the CustomResource itself.

unit tests should not fail after adding the finalizer code. I tested it. There is no need to call `client.Delete()` twice. First call to Delete will trigger the call to remove the finalizer and then CR should automatically be deleted once the finalizer list is empty.

**Comment by droot** [comment]
self-merging it since it was LGTM'd before.

**Pull request merged**